client/js/libs/jquery/inputhistory.js

```diff
@@ -9,48 +9,54 @@
 		history: [],
 		preventSubmit: false
 	};
-	
+
 	$.fn.history = // Alias
 	$.fn.inputhistory = function(options) {
 		options = $.extend(
 			$.inputhistory.defaultOptions,
 			options
 		);
-		
+
 		var self = this;
 		if (self.size() > 1) {
 			return self.each(function() {
 				$(this).history(options);
 			});
 		}
-		
+
 		var history = options.history;
 		history.push("");
-		
+
 		var i = 0;
 		self.on("keydown", function(e) {
 			var key = e.which;
 			switch (key) {
 			case 13: // Enter
-				if (self.val() != "") {
-					i = history.length;
-					history[i - 1] = self.val();
-					history.push("");
-					if (history[i - 1] == history[i - 2]) {
-						history.splice(-2, 1);
-						i--;
+				if (!e.shiftKey) {
+					if (self.val() !== "") {
+						var values = self.val().split("\n");
+						for (var j = 0; j < values.length; j++) {
+							i = history.length;
+							history[i - 1] = values[j];
+							history.push("");
+							if (history[i - 1] === history[i - 2]) {
+                history.splice(-2, 1);
```

Comment:
I wasn't oging to do this, because it's erming's library. But then I realised how much I changed it, so I moved it. And fixed the tabs :-P (it was using spaces, stupid atom).